source/modulo_core/include/modulo_core/concepts.hpp

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+#include <geometry_msgs/msg/accel_stamped.hpp>
+#include <geometry_msgs/msg/pose_stamped.hpp>
+#include <geometry_msgs/msg/transform_stamped.hpp>
+#include <geometry_msgs/msg/twist_stamped.hpp>
+#include <geometry_msgs/msg/wrench_stamped.hpp>
+#include <sensor_msgs/msg/joint_state.hpp>
+#include <std_msgs/msg/bool.hpp>
+#include <std_msgs/msg/float64.hpp>
+#include <std_msgs/msg/float64_multi_array.hpp>
+#include <std_msgs/msg/int32.hpp>
+#include <std_msgs/msg/string.hpp>
+#include <tf2_msgs/msg/tf_message.hpp>
+
+#include "modulo_core/EncodedState.hpp"
+
+#include <state_representation/State.hpp>
+#include <state_representation/space/cartesian/CartesianState.hpp>
+#include <state_representation/space/joint/JointState.hpp>
+
+namespace modulo_core::concepts {
+
+// Data type concepts
+
+template<typename DataT>
+concept StateRepresentationDataT = std::derived_from<DataT, state_representation::State>;
+
+template<typename DataT>
+concept CorePrimitivesT = std::same_as<DataT, int> || std::same_as<DataT, float> || std::same_as<DataT, double>
+    || std::same_as<DataT, bool> || std::same_as<DataT, std::string> || std::same_as<DataT, std::vector<double>>;
+
+template<typename DataT>
+concept CoreDataT = StateRepresentationDataT<DataT> || CorePrimitivesT<DataT>;
+
+template<typename DataT>
+concept CustomDataT = !CoreDataT<DataT>;
+
+// Translation concepts
+
+template<typename DataT>
```

Comment:
Ah also, now I get what was confusing me a bit here. In the translation concepts we should use `MsgT` instead of `DataT` and `TranslatedMsgT` instead of `TranslatedDataT` (and so on) no?

Reply:
I am actually not sure here. I use them to both type check DataT and MsgT. Perhaps we should drop Data/Msg altogether and stay generic by saying TranslatedTypeT and so on. What do you think?